discord/gateway.py

```diff
@@ -142,6 +147,20 @@ def get_payload(self):
             'd': int(time.time() * 1000)
         }
 
+    def ack(self):
+        ack_time = time.perf_counter()
+        self._last_ack = ack_time
+        self.recent_ack_latencies.append(ack_time - self._last_send)
+        # Only the most recent 20 latencies are stored
+        if len(self.recent_ack_latencies) > 20:
+            del self.recent_ack_latencies[0]
+
+        self.latency = self.recent_ack_latencies
+
+        if self.latency[-1] > 10:
+            log.warning(self.behind_msg, self.latency)
```

Comment:
This message doesn't really make sense for what it actually is.

Reply:
Making it clearer:

A latency of 10 means that the ping is 10,000ms which probably means you're blocking somewhere.  For comparison the official client marks your connection red at 250ms. The regular client has a high throughput for events so 10 seconds being behind could just mean there's a lot to process. On the other hand, the voice client websocket is rather low maintenance spending most of its time just heartbeating.